### PR TITLE
Migrating containerapp-compose to containerapp

### PR DESCRIPTION
**Pull request by pauldotyu**
Initial round of migrating containerapp-compose functionality into the containerapp.

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az containerapp compose create`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

**Comment by panchagnula** [comment]
@StrawnSC & @haroonf could you start reviewing this next week?

File: src/containerapp/azext_containerapp/commands.py [hunk 1]
```diff
@@ -174,3 +174,6 @@ def load_command_table(self, _):
         g.custom_command('bind', 'bind_hostname', exception_handler=ex_handler_factory())
         g.custom_command('list', 'list_hostname')
         g.custom_command('delete', 'delete_hostname', confirmation=True, exception_handler=ex_handler_factory())
+
+    with self.command_group('containerapp compose', is_preview=True) as g:
```
Comment (StrawnSC) [inline src/containerapp/azext_containerapp/commands.py:178]: The containerapp extension as a whole is in preview, so I don't think we need this `is_preview=True` here

File: src/containerapp/setup.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
 
 # TODO: Add any additional SDK dependencies here
 DEPENDENCIES = [
-    'azure-cli-core'
+    'azure-cli-core',
+    'pycomposefile>=0.0.29'
 ]
```
Comment (StrawnSC) [inline src/containerapp/setup.py:43]: We'll also need an entry in the history.rst file for this change

**Review by StrawnSC — APPROVED** [review]
Overall this looks good to me. Just a couple of minor comments. 

Once this is ready to go (I see it's still marked as a draft), we may want to change the base branch from Azure:main to one of our containerapp release branches on [this repo](https://github.com/calvinsID/azure-cli-extensions). (Note that the current release branch is [calvinsid:containerapp-0.3.12](https://github.com/Azure/azure-cli-extensions/pull/5350)). We use these branches to batch our containerapp features together in one release

File: src/containerapp/setup.py (hunk 2 above)
Comment (StrawnSC) [reply src/containerapp/setup.py:43]: Also, it looks like some CI steps are failing. We'll need to fix some tests / test recordings to get those passing

File: src/containerapp/azext_containerapp/commands.py (hunk 1 above)
Comment (pauldotyu) [reply src/containerapp/azext_containerapp/commands.py]: Removed `is_preview=True`

File: src/containerapp/setup.py (hunk 2 above)
Comment (pauldotyu) [reply src/containerapp/setup.py:44]: HISTORY.rst file has been updated and CI tests have been re-recorded.

**Comment by pauldotyu** [comment]
> Overall this looks good to me. Just a couple of minor comments.
> 
> Once this is ready to go (I see it's still marked as a draft), we may want to change the base branch from Azure:main to one of our containerapp release branches on [this repo](https://github.com/calvinsID/azure-cli-extensions). (Note that the current release branch is [calvinsid:containerapp-0.3.12](https://github.com/Azure/azure-cli-extensions/pull/5350)). We use these branches to batch our containerapp features together in one release

Thanks for the review @StrawnSC. This is now ready to go.

**Comment by StrawnSC** [comment]
@pauldotyu, yes you can close this PR since this update will be released with the containerapp 0.3.12 version: https://github.com/Azure/azure-cli-extensions/pull/5350. Thanks!

**Comment by pauldotyu** [comment]
Closing this PR. Code has been merged in https://github.com/calvinsID/azure-cli-extensions/pull/155 for release.

**Pull request closed without merging**